learn/guides/managing-dependencies.md

```diff
@@ -12,80 +12,163 @@ redirect_from:
+```bash
+bal push –-repository local <path-to-bala-archive>
+```
 
-### Dependencies.toml
+3. Specify the dependency in `Ballerina.toml`
 
 ```toml
 [[dependency]]
 org = "ballerinax"
-name = "aws.s3"
-version = "0.99.4”
+name = "googleapis.gmail"
+version = "2.1.1"
+repository = "local"
 ```
 
-When you execute `bal build`, the `Dependencies.toml` is created if at least one dependency is referred to its stable version. From the second build onwards, the compiler honors the version declared in the `Dependencies.toml` until you update the file.
+Once the above steps are completed, the dependency will be picked from the local repository when the package is built. 
+During the compilation, the specified version in the `Ballerina.toml` is considered as the minimum required version for that particular dependency 
+and the dependency will be resolved from the local repository. Nonetheless, if the compiler finds a newer patch version in Distribution or Central repositories, 
+then the latest version is always given priority. At this point, the compiler resolves to the latest version and ignores the dependency version in the local repository. 
+
+## Achieving reproducible builds
+
+By default, the compiler always looks up for the latest compatible versions of dependencies in the repositories when building a package. 
+This minimizes the hassle to the package developer in terms of dependency updates since the compiler is smart enough to 
+keep the package updated all the time. However, if you want to repeat a constant behavior to make the build more predictable, 
+then Ballerina facilitates this using offline and sticky modes.
+
+### Sticky mode.
+
+Using the `--sticky` flag with `bal build` will force the compiler to stick to the exact versions locked in the `Dependencies.toml`. 
+In other words, the automatic-update feature is disabled when the `--sticky` flag is provided.
+This can also be set in the `Ballerina.toml` file under `[build-options]` table as follows:
+
+```toml
+[build-options]
+sticky = true
+```
+### Offline mode
 
-When you find a newer version of the `ballerina/io` package, which you want to use in your code, either delete the corresponding dependency declaration from the `Dependencies.toml` file or update the version. If you delete the dependency declaration, then the compiler updates the file with the latest version during the next build.
+Using the` –-offline` flag with `bal build` will run the build offline without connecting to the Ballerina Central. 
+This will save time of the build since the packages are resolved using the Distribution repository and the filesystem cache of the Ballerina Central repository. 
```

Comment:
```suggestion
This will save time of the build since the packages are resolved using the distribution repository and the filesystem cache of the Ballerina Central repository. 
```